hazelcast/src/main/java/com/hazelcast/internal/serialization/InputOutputFactory.java

```diff
@@ -36,7 +36,11 @@ BufferObjectDataInput createInput(byte[] buffer,
                                       InternalSerializationService service,
                                       boolean isCompatibility);
 
-    BufferObjectDataOutput createOutput(int size, InternalSerializationService service);
+    /**
+     * Creating new {@link BufferObjectDataOutput}. The size of the internal buffer is set to initialSize at construction.
+     * When the buffer is too small to accept new bytes the buffer will grow to at least secondInitialSize.
+     */
+    BufferObjectDataOutput createOutput(int initialSize, int secondInitialSize, InternalSerializationService service);
```

Comment:
I would prefer to have two methods here and consequently two constructors of ByteArrayObjectDataOutput.
One is with the old signature, and this one.
This is really a special case factory method where other calls than SQL  are not interested in. 
This way, you don't need `DONT_USE_SECOND_INITIAL_SIZE` as well. This value stay as something private to the implementation. 

After you put the javadoc, I think a more suitable name for `secondInitialSize` is `minGrowthSize`